acme/src/identity/mod.rs

```diff
@@ -32,6 +35,10 @@ pub trait WireIdentityReader {
 
 impl WireIdentityReader for x509_cert::Certificate {
     fn extract_identity(&self) -> RustyAcmeResult<WireIdentity> {
+        let serial_number = self.tbs_certificate.serial_number.as_bytes().to_vec();
+        let serial_number = hex::encode(serial_number);
```

Comment:
Avoid the copy/clone

```suggestion
        let serial_number = hex::encode(self.tbs_certificate.serial_number.as_bytes());
```